lib/compress.js

```diff
@@ -4321,6 +4332,20 @@ merge(Compressor.prototype, {
                 })
             }).optimize(compressor);
         }
+        if (compressor.option("properties") && self.expression instanceof AST_Object) {
```

Comment:
I don't think "properties" is the right option to put this under due to legacy considerations (`a["foo"]` to `a.foo`). I think this new feature belongs in "unsafe" with the other object literal simplifications. Or if you prefer, this can be subject to both "unsafe" *and* "properties".